### DISCUSSION TITLE
Atom value not persistent across the same domain?

### DISCUSSION THREAD

**Discussion by coodoo** [pmndrs/jotai#1564] [Q&A] [answered]
## Summary

Wondering is the atom value supposed to be persistent across the same domain, say we are in `example.com/post/1` and user changed the value of the location to be `example.com/post/2` then hit enter, since it's still the same domain I would assume atom value should be persisted, but it's not.

See [minimal reproduction here](https://codesandbox.io/s/wizardly-joliot-x4zyqe?file=/src/App.js:130-409)

### steps to reproduce
- hit button several times to see counter increase
- place cursor in Browser (the preview window on the right side) location bar and hit enter 
- when the page finished reloading the atom value are reset too, it's now back to 1 instead of keeping prev value
- I would expect the atom value should be persistent across the same domain? 

If this is expected, wondering if there's anyway to work around it? Thanks.

[edit] However when used with `react-router` and navigate with `<Link to='...'>` component to a new location didn't break the atom value, which suggests `client side navigation` works but directly manipulating the browser `location` does not, hth!

**Comment by dai-shi** [comment, marked as answer]
Atom values are store in memory, so browser refresh clears everything. `pushState` works.
You want to use `atomWithStorage` for persistence.
